### PR TITLE
add MODIFY_STACK

### PR DESCRIPTION
**Pull request by bifrost**

### CODE REVIEW

File: src/Reducer.js [hunk 1]
```diff
@@ -246,6 +246,47 @@ function inject(state, action, props, scenes) {
...
+            assert(command.sceneKey || command.index >= 0,
+              `sceneKey or index has to be defined - ${command}`);
+            removedIndex = command.sceneKey ? findIndex(command.sceneKey) : command.index;
+            assert(removedIndex < newChildren.length - 1,
+              `You are not allowed to remove current scene - ${command}`);
+            newChildren.splice(removedIndex, 1);
+            break;
+
+          case ActionConst.ModifyStackTypes.INSERT:
+            const index = command.beforeSceneKey ? findIndex(command.beforeSceneKey) : command.index;
```
Comment (charpeni) [inline src/Reducer.js]: `line 272, col 1, Error - Line 272 exceeds the maximum line length of 100. (max-len)`

File: test/Reducer.modifyStack.test.js [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+      });
+
+      const action = {
+        key: 'privacyPolicy',
+        type: ActionConst.MODIFY_STACK,
+        commands: [
+          { type: ActionConst.ModifyStackTypes.INSERT, sceneKey: 'termsOfService', index: 2 },
+        ],
+      };
+      expect(() => reducer(latestState, action)).to.throw(/You are not allowed change current scene/);
```
Comment (charpeni) [inline test/Reducer.modifyStack.test.js]: `line 289, col 1, Error - Line 289 exceeds the maximum line length of 100. (max-len)`

File: test/Reducer.modifyStack.test.js [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+      latestState = reducer(latestState, {
+        key: 'privacyPolicy',
+        type: ActionConst.PUSH,
+      });
+
+      latestState = reducer(latestState, {
+        key: 'privacyPolicy',
+        type: ActionConst.MODIFY_STACK,
+        commands: [
+          { type: ActionConst.ModifyStackTypes.INSERT, sceneKey: 'termsOfService', beforeSceneKey: 'privacyPolicy' },
```
Comment (charpeni) [inline test/Reducer.modifyStack.test.js]: `line 242, col 1, Error - Line 242 exceeds the maximum line length of 100. (max-len)`

**Comment by aksonov** [comment]
I'm really sorry about missing this PR. Unfortunately I have available time only recently and completed v4 beta based on React Navigation. Feel free to submit PRs for v4 now. I promise I will not miss them again.

**Pull request closed without merging**